### PR TITLE
Refactor e2e to use controller runtime golang client

### PR DESCRIPTION
**Pull request by harche**
This PR removes the dependency on `kubctl` command and instead modifies the e2e tests to use controler 

<details>
  <summary>Sample run on Openshift. Click to expand!</summary>
  
  ```bash
  $ IMG=quay.io/harpatil/instaslice2-controller:2.4 IMG_DMST=quay.io/harpatil/instaslice2-daemonset:2.4 make test-e2e-ocp-emulated
docker build -t quay.io/harpatil/instaslice2-controller:2.4 -f Dockerfile.controller .
WARN[0000] missing "CVE_UPDATES" build argument. Try adding "--build-arg CVE_UPDATES=<VALUE>" to the command line 
[1/2] STEP 1/11: FROM nvcr.io/nvidia/cuda:12.4.1-base-ubi8 AS build
[1/2] STEP 2/11: ARG GOLANG_VERSION=1.22.8
--> Using cache 117a4eb12f9bb787c048c148d79d47435f1e99ece8de7c2400c309d35942dd10
--> 117a4eb12f9b
[1/2] STEP 3/11: RUN yum install -y wget make git gcc
--> Using cache 68c33c95221043454d0ad31f41893e0e74609105ec05f8c526849dc560857aa6
--> 68c33c952210
[1/2] STEP 4/11: RUN set -eux;         arch="$(uname -m)";     case "${arch##*-}" in         x86_64 | amd64) ARCH='amd64' ;;         ppc64el | ppc64le) ARCH='ppc64le' ;;         aarch64) ARCH='arm64' ;;         *) echo "unsupported architecture" ; exit 1 ;;     esac;        wget -nv -O - https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-${ARCH}.tar.gz     | tar -C /usr/local -xz
--> Using cache 5ee198de85335e75b2e53e4cf0916dda44b0e565aeddd42d6dce8564cac6b509
--> 5ee198de8533
[1/2] STEP 5/11: ENV GOPATH=/go
--> Using cache 775820a76743f99c4bcbc52b0c8c1adf946936c6317f2bb05b0a788d55d53e19
--> 775820a76743
[1/2] STEP 6/11: ENV PATH=$GOPATH/bin:/usr/local/go/bin:$PATH
--> Using cache a4e33fe02fadf7d9147a33c510b8682caf8606455fb7e7c05ac6945a49b6ee1d
--> a4e33fe02fad
[1/2] STEP 7/11: WORKDIR /workspace
--> Using cache 154868236a0ff978286183bfa4fbed918e6c09d83a5c28954930c50c94a7f5a9
--> 154868236a0f
[1/2] STEP 8/11: COPY . .
--> ed9c7f934577
[1/2] STEP 9/11: RUN go build -o bin/manager cmd/controller/main.go
--> 0575ea55fb43
[1/2] STEP 10/11: ARG CUDA_VERSION=12.4.1
--> f27a6b5477d0
[1/2] STEP 11/11: ARG BASE_DIST=ubi8
--> 5f2021af8b65
[2/2] STEP 1/10: FROM nvcr.io/nvidia/cuda:12.4.1-base-ubi8
[2/2] STEP 2/10: RUN dnf remove -y cuda-*
--> Using cache 2f5265b49349190c99072c27390d7de67bbe30cd0299684fc2425d29bd9bf97a
--> 2f5265b49349
[2/2] STEP 3/10: ENV NVIDIA_DISABLE_REQUIRE="true"
--> Using cache 969c2e8541ac903605dcc737aff6a9143c06082a3d5816b1d9f98c93e6ec7ada
--> 969c2e8541ac
[2/2] STEP 4/10: ENV NVIDIA_VISIBLE_DEVICES=all
--> Using cache f07027dd038f8d29b29f45a921def355fb75f82effd7756c2e73b8bd5e946112
--> f07027dd038f
[2/2] STEP 5/10: ENV NVIDIA_DRIVER_CAPABILITIES=compute,utility
--> Using cache dadcc1971794b821ebeccfaf64f740499231d0c736001ca7d9d2883901ff5318
--> dadcc1971794
[2/2] STEP 6/10: WORKDIR /
--> Using cache d89ceb5494aab17a288d4f1f4130722da49b9a1a504176804df3aa5b9c44b4b4
--> d89ceb5494aa
[2/2] STEP 7/10: COPY --from=build /workspace/bin/manager .
--> Using cache 192c6e88f85f4ca33886729d976d18e58b2cfce02dcf9244c4fcd8614d00e699
--> 192c6e88f85f
[2/2] STEP 8/10: ARG CVE_UPDATES
--> Using cache 89022a41a84eafed9f40f8631f21c1d8d48fbfc863d3797f1a3557595347654b
--> 89022a41a84e
[2/2] STEP 9/10: RUN if [ -n "${CVE_UPDATES}" ]; then         yum update -y ${CVE_UPDATES} &&         rm -rf /var/cache/yum/*;     fi
--> Using cache cf8023555ef8ed5d5bc888a6c7fda955ebc7509aa6f4014acf2182c0e141e24e
--> cf8023555ef8
[2/2] STEP 10/10: ENTRYPOINT ["/manager"]
--> Using cache 1ddf0753bc6aef13d1a3b7064e38bd78a511f4a1d8ec5565b811ee810f958c3c
[2/2] COMMIT quay.io/harpatil/instaslice2-controller:2.4
--> 1ddf0753bc6a
Successfully tagged quay.io/harpatil/instaslice2-controller:2.4
Successfully tagged quay.io/harpatil/instaslice2-controller:4.0
1ddf0753bc6aef13d1a3b7064e38bd78a511f4a1d8ec5565b811ee810f958c3c
docker build -t quay.io/harpatil/instaslice2-daemonset:2.4 -f Dockerfile.daemonset .
WARN[0000] missing "CVE_UPDATES" build argument. Try adding "--build-arg CVE_UPDATES=<VALUE>" to the command line 
[1/2] STEP 1/14: FROM nvcr.io/nvidia/cuda:12.4.1-base-ubi8 AS build
[1/2] STEP 2/14: ARG GOLANG_VERSION=1.22.2
--> Using cache 117a4eb12f9bb787c048c148d79d47435f1e99ece8de7c2400c309d35942dd10
--> 117a4eb12f9b
[1/2] STEP 3/14: RUN yum install -y wget make git gcc
--> Using cache 618e566922304760fb2231594039f3f757d8a60e672e9c585f71ed1bad5f4927
--> 618e56692230
[1/2] STEP 4/14: RUN set -eux;         arch="$(uname -m)";     case "${arch##*-}" in         x86_64 | amd64) ARCH='amd64' ;;         ppc64el | ppc64le) ARCH='ppc64le' ;;         aarch64) ARCH='arm64' ;;         *) echo "unsupported architecture" ; exit 1 ;;     esac;     wget -nv -O - https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-${ARCH}.tar.gz     | tar -C /usr/local -xz
--> Using cache 7ac4a54435a3b883cb5f1e0ef51342cbfcbb1258d5053907240473127482ed97
--> 7ac4a54435a3
[1/2] STEP 5/14: ENV GOPATH=/go
--> Using cache a1b32df66062a436533fbd05b7a9d37bc59eddd8c1a64af7ea077aadac797e3e
--> a1b32df66062
[1/2] STEP 6/14: ENV PATH=$GOPATH/bin:/usr/local/go/bin:$PATH
--> Using cache 41b79d43f5b784f4379568e8a480778edf4577241822368ee73dba80afe02b4f
--> 41b79d43f5b7
[1/2] STEP 7/14: WORKDIR /workspace
--> Using cache b2acce6ee5475787deba17207d6a35037f16bcc6ea51a0afea41d2a33f572fff
--> b2acce6ee547
[1/2] STEP 8/14: COPY go.mod go.mod
--> Using cache f8497a26f3eca515d7be2328f6e89ad73cfe8b18294a10565e54e0dda53b07ee
--> f8497a26f3ec
[1/2] STEP 9/14: COPY go.sum go.sum
--> Using cache 374182a49b91cec26605648ed1693d2390c4b11ae803a72a607d8a9961e147c1
--> 374182a49b91
[1/2] STEP 10/14: RUN go mod download
--> Using cache d0cf84321aade271cda630a969b8af59a7bb0a9d25e01ec50dc2ee66780c83c8
--> d0cf84321aad
[1/2] STEP 11/14: COPY . .
--> 6b91a41de58f
[1/2] STEP 12/14: RUN go build -o bin/daemonset cmd/daemonset/main.go
--> 6d0c28a9747c
[1/2] STEP 13/14: ARG CUDA_VERSION=12.4.1
--> 8a686f82cf22
[1/2] STEP 14/14: ARG BASE_DIST=ubi8
--> 4e0904060c04
[2/2] STEP 1/10: FROM nvcr.io/nvidia/cuda:12.4.1-base-ubi8
[2/2] STEP 2/10: RUN dnf remove -y cuda-*
--> Using cache 2f5265b49349190c99072c27390d7de67bbe30cd0299684fc2425d29bd9bf97a
--> 2f5265b49349
[2/2] STEP 3/10: ENV NVIDIA_DISABLE_REQUIRE="true"
--> Using cache 969c2e8541ac903605dcc737aff6a9143c06082a3d5816b1d9f98c93e6ec7ada
--> 969c2e8541ac
[2/2] STEP 4/10: ENV NVIDIA_VISIBLE_DEVICES=all
--> Using cache f07027dd038f8d29b29f45a921def355fb75f82effd7756c2e73b8bd5e946112
--> f07027dd038f
[2/2] STEP 5/10: ENV NVIDIA_DRIVER_CAPABILITIES=compute,utility
--> Using cache dadcc1971794b821ebeccfaf64f740499231d0c736001ca7d9d2883901ff5318
--> dadcc1971794
[2/2] STEP 6/10: WORKDIR /
--> Using cache d89ceb5494aab17a288d4f1f4130722da49b9a1a504176804df3aa5b9c44b4b4
--> d89ceb5494aa
[2/2] STEP 7/10: COPY --from=build /workspace/bin/daemonset .
--> Using cache aad28c0ff487a8f8ee391a03514a876dd7a0633841fe37cfce92cd8c0b0eb4c7
--> aad28c0ff487
[2/2] STEP 8/10: ARG CVE_UPDATES
--> Using cache fc34a9fb908328b26cff8552dbe634f4967b8158afc72d594804af2bbe22f4c6
--> fc34a9fb9083
[2/2] STEP 9/10: RUN if [ -n "${CVE_UPDATES}" ]; then         yum update -y ${CVE_UPDATES} &&         rm -rf /var/cache/yum/*;     fi
--> Using cache ac373d5f75f9b0af7dd24b84cf8cfeafb3f82aaf681d7b2df09f37d2fccd770c
--> ac373d5f75f9
[2/2] STEP 10/10: ENTRYPOINT ["/daemonset"]
--> Using cache c089cff2368404db08e00e37a67e07a2589f03ce8a749e3ae908d0836c438b22
[2/2] COMMIT quay.io/harpatil/instaslice2-daemonset:2.4
--> c089cff23684
Successfully tagged quay.io/harpatil/instaslice2-daemonset:2.4
Successfully tagged quay.io/harpatil/instaslice2-daemonset:4.0
c089cff2368404db08e00e37a67e07a2589f03ce8a749e3ae908d0836c438b22
docker push quay.io/harpatil/instaslice2-controller:2.4
Getting image source signatures
Copying blob c411a83bba60 skipped: already exists  
Copying blob 9d63f91420d1 skipped: already exists  
Copying blob 272b5b4abe0e skipped: already exists  
Copying blob 4dc0dfa6555a skipped: already exists  
Copying blob 60ca0de1e02e skipped: already exists  
Copying blob 2efec45cd878 skipped: already exists  
Copying blob a12ce8fe18e8 skipped: already exists  
Copying blob 281e7b273414 skipped: already exists  
Copying blob be3972db4ca0 skipped: already exists  
Copying config 1ddf0753bc done   | 
Writing manifest to image destination
docker push quay.io/harpatil/instaslice2-daemonset:2.4
Getting image source signatures
Copying blob 2efec45cd878 skipped: already exists  
Copying blob 272b5b4abe0e skipped: already exists  
Copying blob 4dc0dfa6555a skipped: already exists  
Copying blob 9d63f91420d1 skipped: already exists  
Copying blob 60ca0de1e02e skipped: already exists  
Copying blob c411a83bba60 skipped: already exists  
Copying blob 313077ae7026 skipped: already exists  
Copying blob a12ce8fe18e8 skipped: already exists  
Copying blob ad0058c1f3db skipped: already exists  
Copying config c089cff236 done   | 
Writing manifest to image destination
export  KUBECTL=oc IMG=quay.io/harpatil/instaslice2-controller:2.4 IMG_DMST=quay.io/harpatil/instaslice2-daemonset:2.4 && \
                hack/deploy-instaslice-emulated-on-ocp.sh
Creating namespace instaslice-system
namespace/instaslice-system created
Installing instaslice CRD
make[1]: Entering directory '/home/harshal/go/src/github.com/openshift/instaslice-operator'
test -s /home/harshal/go/src/github.com/openshift/instaslice-operator/bin/controller-gen && /home/harshal/go/src/github.com/openshift/instaslice-operator/bin/controller-gen --version | grep -q v0.16.4 || \
GOBIN=/home/harshal/go/src/github.com/openshift/instaslice-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.4
/home/harshal/go/src/github.com/openshift/instaslice-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/harshal/go/src/github.com/openshift/instaslice-operator/bin/kustomize build config/crd | oc apply -f -
customresourcedefinition.apiextensions.k8s.io/instaslices.inference.redhat.com created
make[1]: Leaving directory '/home/harshal/go/src/github.com/openshift/instaslice-operator'
Deploying Instaslice controller-manager
make[1]: Entering directory '/home/harshal/go/src/github.com/openshift/instaslice-operator'
test -s /home/harshal/go/src/github.com/openshift/instaslice-operator/bin/controller-gen && /home/harshal/go/src/github.com/openshift/instaslice-operator/bin/controller-gen --version | grep -q v0.16.4 || \
GOBIN=/home/harshal/go/src/github.com/openshift/instaslice-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.4
/home/harshal/go/src/github.com/openshift/instaslice-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
cd config/manager && /home/harshal/go/src/github.com/openshift/instaslice-operator/bin/kustomize edit set image controller=quay.io/harpatil/instaslice2-controller:2.4
oc apply -f config/rbac/instaslice-operator-scc.yaml
securitycontextconstraints.security.openshift.io/instaslice-operator-scc created
oc apply -f config/rbac/openshift_scc_cluster_role_binding.yaml
clusterrolebinding.rbac.authorization.k8s.io/use-scc-instaslice-operator created
oc apply -f config/rbac/openshift_cluster_role.yaml
clusterrole.rbac.authorization.k8s.io/system:openshift:scc:instaslice-operator-scc unchanged
/home/harshal/go/src/github.com/openshift/instaslice-operator/bin/kustomize build config/emulator | sed -e "s|<IMG_DMST>|quay.io/harpatil/instaslice2-daemonset:2.4|g" | oc apply -f -
Warning: resource namespaces/instaslice-system is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
namespace/instaslice-system configured
customresourcedefinition.apiextensions.k8s.io/instaslices.inference.redhat.com configured
serviceaccount/instaslice-operator-controller-manager created
role.rbac.authorization.k8s.io/instaslice-operator-leader-election-role created
clusterrole.rbac.authorization.k8s.io/instaslice-operator-manager-role created
clusterrole.rbac.authorization.k8s.io/instaslice-operator-metrics-reader created
clusterrole.rbac.authorization.k8s.io/instaslice-operator-proxy-role created
rolebinding.rbac.authorization.k8s.io/instaslice-operator-leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/instaslice-operator-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/instaslice-operator-proxy-rolebinding created
service/instaslice-operator-controller-manager-metrics-service created
service/instaslice-operator-webhook-service created
deployment.apps/instaslice-operator-controller-manager created
certificate.cert-manager.io/instaslice-operator-serving-cert created
issuer.cert-manager.io/instaslice-operator-selfsigned-issuer created
mutatingwebhookconfiguration.admissionregistration.k8s.io/instaslice-operator-mutating-webhook-configuration created
make[1]: Leaving directory '/home/harshal/go/src/github.com/openshift/instaslice-operator'
pod/instaslice-operator-controller-manager-798b7cdd-57c7h condition met
Patching node ip-10-0-76-126.us-east-2.compute.internal
node/ip-10-0-76-126.us-east-2.compute.internal patched (no change)
export KUBECTL=oc IMG=quay.io/harpatil/instaslice2-controller:2.4 IMG_DMST=quay.io/harpatil/instaslice2-daemonset:2.4 && \
ginkgo -v --json-report=report.json --junit-report=report.xml --timeout 20m  ./test/e2e
Ginkgo detected a version mismatch between the Ginkgo CLI and the version of Ginkgo imported by your packages:
  Ginkgo CLI Version:
    2.15.0
  Mismatched package versions found:
    2.22.0 used by e2e

  Ginkgo will continue to attempt to run but you may see errors (including flag
  parsing errors) and should either update your go.mod or your version of the
  Ginkgo CLI to match.

  To install the matching version of the CLI run
    go install github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file.  Alternatively you can use
    go run github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file to invoke the matching version of the
  Ginkgo CLI.

  If you are attempting to test multiple packages that each have a different
  version of the Ginkgo library with a single Ginkgo CLI that is currently
  unsupported.
  
  Starting instaslice-operator suite
Running Suite: e2e suite - /home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e
=================================================================================================
Random Seed: 1733766584

Will run 11 of 11 specs
------------------------------
[BeforeSuite] 
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:95
  cri-bin: docker
  kubectl-bin: kubectl
  namespace: instaslice-system
  emulated: true
  node-name: kind-e2e-control-plane
  controller-image: quay.io/harpatil/instaslice2-controller:2.4
  daemonset-image: quay.io/harpatil/instaslice2-daemonset:2.4
[BeforeSuite] PASSED [0.241 seconds]
------------------------------
controller Operator should create a pod with no requests and check if finalizer exists
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:167
• [10.567 seconds]
------------------------------
controller Operator should create a pod with no requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:194
• [0.369 seconds]
------------------------------
controller Operator should create a pod with small requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:220
• [0.380 seconds]
------------------------------
controller Operator should create a pod with large memory requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:246
• [60.268 seconds]
------------------------------
controller Operator should create a pod with large cpu requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:272
• [60.230 seconds]
------------------------------
controller Operator should create a deployment and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:298
• [5.334 seconds]
------------------------------
controller Operator should create a statefulset and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:337
• [0.275 seconds]
------------------------------
controller Operator should create a job and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:376
• [0.313 seconds]
------------------------------
controller Operator should verify all MIG slice capacities are as expected before submitting pods
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:417
• [0.653 seconds]
------------------------------
controller Operator should verify the existence of pod allocations
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:458
• [31.068 seconds]
------------------------------
controller Operator should verify that the Kubernetes node has the specified resource and matches total GPU memory
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:510
  STEP: Verifying that node has custom resource instaslice.redhat.com/accelerator-memory-quota @ 12/09/24 12:52:36.587
• [0.197 seconds]
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --json-report --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.005 seconds]
------------------------------

Ran 11 of 11 Specs in 169.897 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 2m51.852150727s
Test Suite Passed
```

</details>

### CODE REVIEW

**Comment by harche** [comment]
/hold 

I haven't tested this on a `kind` cluster yet.

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -101,394 +133,351 @@ var _ = BeforeSuite(func() {
...
 				}
+			})
 
-				var result struct {
-					Items []map[string]interface{} `json:"items"`
-				}
-				err = json.Unmarshal(output, &result)
+			Eventually(func() error {
+				err := k8sClient.Get(ctx, client.ObjectKey{Name: pod.Name, Namespace: pod.Namespace}, pod)
 				if err != nil {
```
Comment (rphillips) [inline test/e2e/e2e_test.go:180]: could be cleaned up with an Expect(err) as well

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -92,6 +99,31 @@ var _ = BeforeSuite(func() {
...
+
+	//+kubebuilder:scaffold:scheme
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	ctx = context.TODO()
+
+	node, err := getNodeName(kubectlBin, controllerManagerLabel)
+	if node != "" && err == nil {
```
[the remark this answers is not in the document]
Comment (harche) [reply test/e2e/e2e_test.go:120]: yes, missed that, thanks.

**Comment by harche** [comment]
> /hold
> 
> I haven't tested this on a `kind` cluster yet.

/hold cancel 

<details>
  <summary>Sample run using Kind. Click to expand!</summary>
  

```bash
  Starting instaslice-operator suite
Running Suite: e2e suite - /home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e
=================================================================================================
Random Seed: 1733846544

Will run 11 of 11 specs
------------------------------
[BeforeSuite] 
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:94
  cri-bin: docker
  kubectl-bin: kubectl
  namespace: instaslice-system
  emulated: true
  node-name: kind-e2e-control-plane
  controller-image: quay.io/harpatil/instaslice2-controller:4.0
  daemonset-image: quay.io/harpatil/instaslice2-daemonset:4.0
[BeforeSuite] PASSED [0.007 seconds]
------------------------------
controller Operator should create a pod with no requests and check if finalizer exists
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:167
• [20.081 seconds]
------------------------------
controller Operator should create a pod with no requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:194
• [5.062 seconds]
------------------------------
controller Operator should create a pod with small requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:220
• [5.059 seconds]
------------------------------
controller Operator should create a pod with large memory requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:246
• [60.055 seconds]
------------------------------
controller Operator should create a pod with large cpu requests and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:271
• [60.130 seconds]
------------------------------
controller Operator should create a deployment and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:297
• [5.135 seconds]
------------------------------
controller Operator should create a statefulset and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:336
• [5.098 seconds]
------------------------------
controller Operator should create a job and check the allocation in instaslice object
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:375
• [5.092 seconds]
------------------------------
controller Operator should verify all MIG slice capacities are as expected before submitting pods
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:416
• [0.046 seconds]
------------------------------
controller Operator should verify the existence of pod allocations
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:455
• [30.333 seconds]
------------------------------
controller Operator should verify that the Kubernetes node has the specified resource and matches total GPU memory
/home/harshal/go/src/github.com/openshift/instaslice-operator/test/e2e/e2e_test.go:507
  STEP: Verifying that node has custom resource instaslice.redhat.com/accelerator-memory-quota @ 12/10/24 11:05:42.861
• [0.021 seconds]
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --json-report --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.002 seconds]
------------------------------

Ran 11 of 11 Specs in 196.120 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 3m18.198726329s
Test Suite Passed
```
</details>

**Pull request merged**